DQM/HLTEvF/plugins/LumiMonitor.cc

```diff
@@ -0,0 +1,212 @@
+#include "DQM/HLTEvF/plugins/LumiMonitor.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DQM/TrackingMonitor/interface/GetLumi.h"
+
+
+// -----------------------------
+//  constructors and destructor
+// -----------------------------
+
+LumiMonitor::LumiMonitor( const edm::ParameterSet& iConfig ) : 
+  conf_(iConfig) 
+  , doPixelLumi_                  ( conf_.getParameter<bool>  ("doPixelLumi")                 )
+  , useBPixLayer1_                ( conf_.getParameter<bool>  ("useBPixLayer1")               )
+  , minNumberOfPixelsPerCluster_  ( conf_.getParameter<int>   ("minNumberOfPixelsPerCluster") )
+  , minPixelClusterCharge_        ( conf_.getParameter<double>("minPixelClusterCharge")       )
+  
+{
+
+  folderName_ = conf_.getParameter<std::string>("FolderName");
+
+  pixelClustersToken_ = mayConsume<edmNew::DetSetVector<SiPixelCluster> >(conf_.getParameter<edm::InputTag>("pixelClusters"));
+  lumiscalersToken_   = mayConsume<LumiScalersCollection>(conf_.getParameter<edm::InputTag>("scalers"));
+
+  numberOfPixelClustersVsLS_   = NULL;
+  numberOfPixelClustersVsLumi_ = NULL;
+  lumiVsLS_                    = NULL;
+  pixelLumiVsLS_               = NULL;
+  pixelLumiVsLumi_             = NULL;
+
+  if(useBPixLayer1_) 
+    lumi_factor_per_bx_ = GetLumi::FREQ_ORBIT * GetLumi::SECONDS_PER_LS / GetLumi::XSEC_PIXEL_CLUSTER  ;
+  else
+    lumi_factor_per_bx_ = GetLumi::FREQ_ORBIT * GetLumi::SECONDS_PER_LS / GetLumi::rXSEC_PIXEL_CLUSTER  ;
+
+
+}
+
+LumiMonitor::~LumiMonitor() {
+
+}
+
+void LumiMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+				 edm::Run const        & iRun,
+				 edm::EventSetup const & iSetup) 
+{  
+  
+  std::string histname, histtitle;
+
```

Comment:
I would prefer not to keep a copy of the configuration as a data member, and not to extract the parameters at every run boundary.
Could you extract all the values from the configuration in the constructor, and store them in data members?